test/lib/test_helper.rb

```diff
@@ -20,6 +20,16 @@ def work(time = 2, options = {})
     sleep time
     w.shutdown
   end
+
+  def with_worker(options = {})
+    w = Workhorse::Worker.new(options)
+    begin
+      w.start
```

Comment:
I would place this outside of the `begin` block. When an exception happens during startup, we don't want to call `shutdown`.